### PR TITLE
chore: Adding wsl linter

### LINKED ISSUE #655: Adding wsl linter

### PR DESCRIPTION
**Pull request by bansaltushar014**
## Description

This PR is related to WSL Linter. Have enabled WSL Linter and made changes as per requirement. 

## Related issue(s)

- Fixes #655

### CODE REVIEW

File: crypto/bls/private_key.go [hunk 1]
```diff
@@ -48,17 +47,14 @@ func PrivateKeyFromString(text string) (*PrivateKey, error) {
...
 // Based on https://datatracker.ietf.org/doc/html/draft-irtf-cfrg-bls-signature-04#section-2.3
+//nolint:all
 func KeyGen(ikm, keyInfo []byte) (*PrivateKey, error) {
 	// L is `ceil((3 * ceil(log2(r))) / 16) = 48`,
 	//    where `r` is the order of the BLS 12-381 curve
 	//    r:  0x73eda753 299d7d48 3339d808 09a1d805 53bda402 fffe5bfe ffffffff 00000001
 	// 	  https://datatracker.ietf.org/doc/html/draft-yonezawa-pairing-friendly-curves-02#section-4.2.2
 	//
 
-	if len(ikm) < 32 {
```
Comment (kehiy) [inline crypto/bls/private_key.go:58]: revert this line, it fixes your pipeline issue.

**Review by kehiy — CHANGES_REQUESTED** [review]

**Comment by bansaltushar014** [comment]
Sure, Doing it.

**Comment by kehiy** [comment]
@bansaltushar014 thanks, you can use `make unit_test` to run tests locally.

**Comment by bansaltushar014** [comment]
Done with change and commit, All test cases are success.

File: bin/golangci-lint
Comment (kehiy) [inline bin/golangci-lint:1]: @bansaltushar014 also removed this binary.

**Review by kehiy — CHANGES_REQUESTED** [review]

File: bin/golangci-lint
Comment (bansaltushar014) [reply bin/golangci-lint:1]: Done.

**Review by kehiy — APPROVED** [review]
LGTM
@bansaltushar014 thank you.

File: sync/handler_transactions.go [hunk 2]
```diff
@@ -21,10 +21,11 @@ func (handler *transactionsHandler) ParseMessage(m message.Message, _ peer.ID) e
 	handler.logger.Trace("parsing Transactions message", "message", msg)
 
 	for _, trx := range msg.Transactions {
+		//nolint:all
 		if err := handler.state.AddPendingTx(trx); err != nil {
 			handler.logger.Debug("cannot append transaction", "tx", trx, "error", err)
 
-			// TODO: set peer as bad peer?
+			/** TODO: set peer as bad peer? **/
```
Comment (kehiy) [inline sync/handler_transactions.go:28]: revert this line, it make some codecov errors.

**Review by kehiy — CHANGES_REQUESTED** [review]
check the GUI errors, you have some missing wsl errors.

File: sync/handler_transactions.go (hunk 2 above)
Comment (bansaltushar014) [reply sync/handler_transactions.go]: Reverted.

File: cmd/cmd.go [hunk 3]
```diff
@@ -160,6 +164,7 @@ func PrintErrorMsgf(format string, a ...interface{}) {
 		// Print error msg with red color
 		format = fmt.Sprintf("\033[31m[ERROR] %s\033[0m", format)
 	}
+
```
Comment (b00f) [inline cmd/cmd.go:167]: The function has no return. Why wsl add a line here

File: cmd/cmd.go [hunk 4]
```diff
@@ -168,6 +173,7 @@ func PrintSuccessMsgf(format string, a ...interface{}) {
 		// Print successful msg with green color
 		format = fmt.Sprintf("\033[32m%s\033[0m", format)
 	}
+
```
Comment (b00f) [inline cmd/cmd.go:176]: Same here,

**Comment by b00f** [comment]
@bansaltushar014

The `wsl` linter has some issues, and I'd like to point out some of them here:

1. It is not compatible with [gofumpt](https://github.com/mvdan/gofumpt); some parts will be overwritten by `gofumpt`.
2. Some [rules](https://github.com/bombsimon/wsl/blob/master/doc/rules.md) are not configurable.
3. It adds extra lines in some places that appear redundant.

Overall, while it may improve code readability in some cases, its rules are generally not helpful for our purposes. Therefore, we will not be supporting it.

Thank you for your pull request, but I am going to close it. 
Feel free to choose another issue to work on.

**Pull request closed without merging**